src/transformers/models/fnet/convert_fnet_original_flax_checkpoint_to_pytorch.py

```diff
@@ -0,0 +1,155 @@
+    )
+    new_state_dict["fnet.embeddings.LayerNorm.weight"] = torch.tensor(
+        pretrained_model_params["encoder"]["embedder"]["layer_norm"]["scale"]
+    )
+    new_state_dict["fnet.embeddings.LayerNorm.bias"] = torch.tensor(
+        pretrained_model_params["encoder"]["embedder"]["layer_norm"]["bias"]
+    )
+
+    # Encoder Layers
+    for layer in range(config.num_hidden_layers):
+        new_state_dict[f"fnet.encoder.layer.{layer}.fourier.output.LayerNorm.weight"] = torch.tensor(
+            pretrained_model_params["encoder"][f"encoder_{layer}"]["mixing_layer_norm"]["scale"]
+        )
+        new_state_dict[f"fnet.encoder.layer.{layer}.fourier.output.LayerNorm.bias"] = torch.tensor(
+            pretrained_model_params["encoder"][f"encoder_{layer}"]["mixing_layer_norm"]["bias"]
+        )
+
+        new_state_dict[f"fnet.encoder.layer.{layer}.intermediate.dense.weight"] = torch.tensor(
+            pretrained_model_params["encoder"][f"feed_forward_{layer}"]["intermediate"]["kernel"]
+        ).T
+        new_state_dict[f"fnet.encoder.layer.{layer}.intermediate.dense.bias"] = torch.tensor(
+            pretrained_model_params["encoder"][f"feed_forward_{layer}"]["intermediate"]["bias"]
+        )
+
+        new_state_dict[f"fnet.encoder.layer.{layer}.output.dense.weight"] = torch.tensor(
+            pretrained_model_params["encoder"][f"feed_forward_{layer}"]["output"]["kernel"]
+        ).T
+        new_state_dict[f"fnet.encoder.layer.{layer}.output.dense.bias"] = torch.tensor(
+            pretrained_model_params["encoder"][f"feed_forward_{layer}"]["output"]["bias"]
+        )
+
+        new_state_dict[f"fnet.encoder.layer.{layer}.output.LayerNorm.weight"] = torch.tensor(
+            pretrained_model_params["encoder"][f"encoder_{layer}"]["output_layer_norm"]["scale"]
+        )
+        new_state_dict[f"fnet.encoder.layer.{layer}.output.LayerNorm.bias"] = torch.tensor(
+            pretrained_model_params["encoder"][f"encoder_{layer}"]["output_layer_norm"]["bias"]
+        )
+
+    # Pooler Layers
+    new_state_dict["fnet.pooler.dense.weight"] = torch.tensor(pretrained_model_params["encoder"]["pooler"]["kernel"]).T
+    new_state_dict["fnet.pooler.dense.bias"] = torch.tensor(pretrained_model_params["encoder"]["pooler"]["bias"])
+
+    # Masked LM Layers
+    new_state_dict["cls.predictions.transform.dense.weight"] = torch.tensor(
+        pretrained_model_params["predictions_dense"]["kernel"]
+    ).T
+    new_state_dict["cls.predictions.transform.dense.bias"] = torch.tensor(
+        pretrained_model_params["predictions_dense"]["bias"]
+    )
+    new_state_dict["cls.predictions.transform.LayerNorm.weight"] = torch.tensor(
+        pretrained_model_params["predictions_layer_norm"]["scale"]
+    )
+    new_state_dict["cls.predictions.transform.LayerNorm.bias"] = torch.tensor(
+        pretrained_model_params["predictions_layer_norm"]["bias"]
+    )
+    new_state_dict["cls.predictions.decoder.weight"] = torch.tensor(
```

Comment:
actually I don't think that this is needed as the output word embeddings are tied to the input word embeddings anyways